clients/ValkeyClients.md

```diff
@@ -0,0 +1,174 @@
+| **Read from replica**                        |     Yes      |    Yes    |
+| **Exponential backoff to prevent storm**     |     Yes      |    Yes    |
+| **Valkey version compatibility**             |     7.2      |    7.2    |
+| **PubSub state restoration**                 |     Yes      |    No     |
+| **Cluster Scan**                             |     Yes      |    No     |
+| **Latency-Based Read from Replica**          |     No       |    No     |
+| **Client Side Caching**                      |     No       |    No     |
+
+## Java
+
+- **valkey-glide**
+  - GitHub: [valkey-glide](https://github.com/valkey-io/valkey-glide/tree/main/java)
+  - Installation: Available via Maven and Gradle
+  - Description: Valkey GLIDE is designed for reliability, optimized performance, and high-availability, for Valkey and Redis OSS based applications. GLIDE is a multi language client library, written in Rust with programming language bindings, such as Java, node.js and Python.
+
+- **Valkey-Java**
+  - GitHub: [valkey-java](https://github.com/valkey-io/valkey-java)
+  - Installation: Available via Maven and Gradle
+  - Description: valkey-java is Valkey's Java client, derived from Jedis fork, dedicated to maintaining simplicity and high performance.
+
+<!--- This heading should not be included in the TOC --->
+### Feature Comparison Table
+{: .no_toc }
+
+| Feature                                      | valkey-glide | valkey-java |
+|----------------------------------------------|:------------:|:-----------:|
+| **Read from replica**                        |     Yes      |     No      |
+| **Exponential backoff to prevent storm**     |     Yes      |     Yes     |
+| **Valkey version compatibility**             |     7.2      |     7.2     |
+| **PubSub state restoration**                 |     Yes      |     No      |
+| **Cluster Scan**                             |     Yes      |     No      |
+| **Latency-Based Read from Replica**          |     No       |     No      |
+| **Client Side Caching**                      |     No       |     No      |
+
+
+## Go
+
+- **valkey-go**
+  - GitHub: [go-valkey-go](https://github.com/valkey-io/valkey-go)
+  - Installation: TBD
+  - Description: A fast Golang Valkey client that does auto pipelining and supports server-assisted client-side caching.
+
+<!--- This heading should not be included in the TOC --->
+### Feature Comparison Table
+{: .no_toc }
+
+| Feature                                      | valkey-go  |
+|----------------------------------------------|:----------:|
+| **Read from replica**                        |     Yes    |
+| **Exponential backoff to prevent storm**     |     Yes    |
+| **Valkey version compatibility**             |     7.2    |
+| **PubSub state restoration**                 |     No     |
+| **Cluster Scan**                             |     No     |
+| **Latency-Based Read from Replica**          |     No     |
+| **Client Side Caching**                      |     No     |
```

Comment:
Hi @madolson, @asafpamzn thank you for mentioning valkey-go. The current valkey-go should be like this:

```suggestion
| **Exponential backoff to prevent storm**     |     No     |
| **Valkey version compatibility**             |     7.2    |
| **PubSub state restoration**                 |     Yes    |
| **Cluster Scan**                             |     No     |
| **Latency-Based Read from Replica**          |     No     |
| **Client Side Caching**                      |     Yes    |
```

valkey-go doesn't do exponential backoff on reconnections but it makes sure there is only one TCP handshake to the same destination at a time. Exponential backoff can stil be added in the next version.

Reply:
Some backoff between retries is good, but exponential backoff is not necessarily wanted in all situations.

We don't use it much in Ericsson (at least not in code I've encountered). If a node or peer is down for a long time and you double the wait between each retry, you may end up waiting extremely long before trying to reconnect when it's up again. If used, it should be at least capped to some maximum wait (a few seconds) between retries IMO.

Reply:
Maybe we should clarify it to something like, "smart connection backoff"?

Reply:
Yeah.

But it depends on the setup too. In our systems, we don't have that many clients connected, but we have very much traffic on few clients.

Reply:
I agree that Jitter is good enough in most cases. I'm fine with this change.